### PR TITLE
Pickup kind commit that fixes node image building

### PR DESCRIPTION
**Pull request by msau42**
Tested new kind version in https://github.com/kubernetes-csi/csi-driver-host-path/pull/106

Fixes: https://github.com/kubernetes-csi/csi-release-tools/issues/38

### CODE REVIEW

File: config/jobs/kubernetes-csi/gen-jobs.sh [hunk 1]
```diff
@@ -506,6 +506,9 @@ for tests in non-alpha alpha; do
       args:
       - ./.prow.sh
       env:
+      # TODO: change to v0.6.0
```
Comment (davidz627) [inline config/jobs/kubernetes-csi/gen-jobs.sh]: why is there a TODO on all of these

Comment (msau42) [reply config/jobs/kubernetes-csi/gen-jobs.sh]: Using a specific commit of kind is a workaround until kind releases a new version. Updated with bug reference

**Comment by pohly** [comment]
As a stop-gap measure this is okay, but I think it's cleaner to let individual projects pick the version of kind that works for them and what they need kind for.

**Comment by msau42** [comment]
> As a stop-gap measure this is okay, but I think it's cleaner to let individual projects pick the version of kind that works for them and what they need kind for.

I actually prefer the model where all versions are defined in configs and the prow.sh script itself has as few versions encoded in it as possible. It makes it a lot simpler to apply changes consistently across all repos. I don't see a strong use case for having different repos use different versions and it has not come up as a need for us yet.

If different repos really want different configs, then they can specify that here.

**Comment by pohly** [comment]
Michelle Au <notifications@github.com> writes:
> I actually prefer the model where all versions are defined in configs
> and the prow.sh script itself has as few versions encoded in it as
> possible. It makes it a lot simpler to apply changes consistently
> across all repos.

There are two sides of the coin here. When moving configuration into the
job definition, then it becomes harder to keep the configuration
consistent with the source code.

If you ever get into a situation where source code and configuration
need to be changed in lock-step, then merging is blocked: the source
code might fail with the current test configuration and thus cannot be
merged. The test configuration in test-infra cannot be merged either
because then other PRs in the project fail.

It also becomes harder to do maintenance releases. Old branches of a
project may need a different test configuration than master, so you
would have to start defining different jobs for different branches.

**Pull request merged**